src/consensus/aft/raft.h

```diff
@@ -1344,7 +1334,10 @@ namespace aft
               }
               else
               {
-                state->view_history.update(prev_lci + 1, ds->get_term());
+                // NB: This is only safe as long as AppendEntries only contain a
+                // single term. If they cover multiple terms, then we need to
+                // know our previous signature locally.
+                state->view_history.update(r.prev_idx + 1, ds->get_term());
```

Comment:
`Multi-term divergence` committable suffix test found the following buggy situation with the old code:

```
B: 1.1, 1.2, 1.3, 1.4, 7.5, 7.6, 7.7, 7.8, 8.9, 9.10
A: 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 3.7, 5.8
B->A: AE (1.4, 7.6], term=7, prev_term=1
```

Node A has cleared their `committable_indices`, so no longer considers the shared entry 1.4 as committable. So when deserialising this signature, the LCI is incorrect, and can't be used to work out where this signature's term starts from.

As long as we don't have multi-term AEs, we can use `r.prev_idx` instead - the LCI (and a persistent history of signatures) is only necessary when the AE might end in a different term.